ast_canopy/cpp/cmake/ast_canopy_llvm.cmake

```diff
@@ -0,0 +1,51 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+function(ast_canopy_check_llvm_linkage library_target)
+  if(NOT DEFINED LLVM_LINKAGE)
+    message(STATUS "LLVM_LINKAGE is not defined, using SHARED")
+    set(LLVM_LINKAGE "SHARED")
```

Comment:
My 2c: we probably shouldn't even bother supporting linking to LLVM shared libraries. It isn't safe to mix different LLVM library versions in a process without statically linking and its such a common use case these days that we should probably just move to always statically linking. Conda-forge is generally moving to the same after finding countless issues over time with LLVM shared linkage.